badger/cmd/write_bench.go

```diff
@@ -415,3 +419,38 @@ func dropAll(c *y.Closer, db *badger.DB) {
 		}
 	}
 }
+
+func dropPrefix(c *y.Closer, db *badger.DB) {
+	defer c.Done()
+
+	dropPeriod, err := time.ParseDuration(dropPrefixPeriod)
+	y.Check(err)
+	if dropPeriod == 0 {
+		return
+		log.Println("Exit from drop prefix")
+	}
+
+	t := time.NewTicker(dropPeriod)
+	defer t.Stop()
+	for {
+		select {
+		case <-c.HasBeenClosed():
+			return
+		case <-t.C:
+			fmt.Println("[DropPrefix] Started")
+			err := db.DropAll()
```

Comment:
You're dropping all the data. DropAll followed by drop prefix wouldn't do anything.

Reply:
fixed this